src/languageServerInstaller.ts

```diff
@@ -168,14 +204,41 @@ export class LanguageServerInstaller {
 		});
 	}
 
+	verify(release: { builds?: any[]; version: any; shasums?: any; shasums_signature?: any; }, pkg: string, buildName: string) {
+		return new Promise<void>((resolve, reject) => {
+			const hash = crypto.createHash('sha256');
+			const pkgStream = fs.createReadStream(pkg);
+
+			pkgStream.on('data', (data) => {
+				hash.update(data);
+			});
+
+			let shasumResponse = "";
+			https.get(`${releasesUrl}/${release.version}/${release.shasums}`, (response) => {
+				response.on('data', (data) => {
+					shasumResponse += data;
+				});
+				response.on('end', () => {
+					const shasumLine = shasumResponse.split(`\n`).find(line => line.includes(buildName));
+					if (!shasumLine) {
+						return reject(`Install error: no matching SHA sum for ${buildName}`);
+					}
+					const shasum = shasumLine.split(" ")[0];
+					if (hash.digest('hex') !== shasum) {
```

Comment:
practically speaking this likely always works but I believe `hash.digest()` should not be called until `pkgStream.on('end')` fires. This can be worked into a follow up PR though without going too far down the rabbit hole, but a little hashing promise helper could help

```
function hash(file) {
	return new Promise((resolve, reject) => {
		let hash = crypto.createHash('sha256');
		fs.createReadStream(file)
			.on('data', data => hash.update(data))
			.on('end', () => resolve(hash.digest('hex')))
			.on('error', reject);
	});
}
```
Then you could avoid the "outer promise" and have `verify` be written as
```
function verify(...) {
    return hash(pkg)
        .then(() => /* http request stuff, where you can then safely call hash.digest */)
}
```

Reply:
This is a good idea that I'll come back to when we add the signature verification.